### PR TITLE
Zap Fix

### LINKED ISSUE #2768: Spike - fix production owasp scan reporting
**Description:**
Per [2766](https://github.com/raft-tech/TANF-app/issues/2766#issuecomment-1841558675), the failure of the owasp scan job in the pipeline means the scan report is not being uploaded to the django admin console. The report should be uploaded any time the scan completes, regardless of findings that cause the scan to fail.

**Deliverable(s):** 

- [ ] Investigate allowing the zap scan report to be uploaded to the django admin console, whether the scan contains errors or not
    - 4 hour timebox

### PR DESCRIPTION
**Pull request by elipe17**
## Summary of Changes
_Provide a brief summary of changes_
Pull request closes #2768 

## How to Test
1. Kick off the `build-and-test-all` CI task
2. Or, look back at the past tasks and verify that the `develop` DAC has the reports from the same timestamps

## Deliverables
_More details on how deliverables herein are assessed included [here](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
+ [x] Zap Scan completes
+ [x] Zap Scan report successfully uploads to DAC
+ [x] Pipeline passes

### [Deliverable 2: Tested Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
  + [x] If this PR introduces backend code changes, are they meaningfully tested?
  + [x] If this PR introduces frontend code changes, are they meaningfully tested?
+ Are code coverage minimums met?
  + [x] Frontend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)
  + [x] Backend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)

### [Deliverable 3: Properly Styled Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)

+ [x] Are backend code style checks passing on CircleCI?
+ [x] Are frontend code style checks passing on CircleCI?
+ [x] Are code maintainability principles being followed?

### [Deliverable 4: Accessible](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)

+ [ ] Does this PR complete the epic? 
+ [ ] Are links included to any other gov-approved PRs associated with epic?
+ [ ] Does PR include documentation for Raft's a11y review? 
+ [ ] Did automated and manual testing with `iamjolly` and `ttran-hub` using Accessibility Insights reveal any errors introduced in this PR?

### [Deliverable 5: Deployed](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)

+ [ ] Was the code successfully deployed via automated CircleCI process to development on Cloud.gov?

### [Deliverable 6: Documented](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)

+ [x] Does this PR provide background for why coding decisions were made?
+ [ ] If this PR introduces backend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces frontend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces dependencies, are their licenses documented?
+ [x] Can reviewer explain and take ownership of these elements presented in this code review?

### [Deliverable 7: Secure](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)

+ [x] Does the OWASP Scan pass on CircleCI?
+ [ ] Do manual code review and manual testing detect any new security issues?
+ [ ] If new issues detected, is investigation and/or remediation plan documented? 

### [Deliverable 8: User Research](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
+ [ ] the purpose of the research
+ [ ] methods used to conduct the research 
+ [ ] who participated in the research
+ [ ] what was tested and how
+ [ ] impact of research on TDP
+ [ ] (_if applicable_) final design mockups produced for TDP development

### CODE REVIEW

File: .circleci/owasp/jobs.yml [hunk 1]
```diff
@@ -88,20 +88,9 @@
...
             # Evaluate the full command before passing it in so it doesn't
             # get improperly interpolated by Cloud.gov.
-            CMD="python manage.py process_owasp_scan ${CMD_ARGS[@]}"
+            CMD="python manage.py process_owasp_scan $CIRCLE_BUILD_NUM --backend-pass-count ${ZAP_BACKEND_PASS_COUNT:-0} --backend-warn-count ${ZAP_BACKEND_WARN_COUNT:-0} --backend-fail-count ${ZAP_BACKEND_FAIL_COUNT:-0} --frontend-pass-count ${ZAP_FRONTEND_PASS_COUNT:-0} --frontend-warn-count ${ZAP_FRONTEND_WARN_COUNT:-0} --frontend-fail-count ${ZAP_FRONTEND_FAIL_COUNT:-0} --project-slug $PROJECT_SLUG"
+
             # Submit a CF Task for execution that will run the necessary command
-            cf run-task tdp-backend-<< parameters.target_env >> \
-              --command "$CMD" \
-              --name nightly-owasp-scan
+            cf run-task tdp-backend-<< parameters.target_env >> --wait --command "$CMD" --name nightly-owasp-scan
```
Comment (andrew-jameson) [inline .circleci/owasp/jobs.yml]: Can we nix the `--wait` command? I'm not sure it was effective in my testing.

File: .circleci/owasp/jobs.yml [hunk 2]
```diff
@@ -88,20 +88,9 @@
...
-              --backend-fail-count ${ZAP_BACKEND_FAIL_COUNT:-0}
-              --frontend-pass-count ${ZAP_FRONTEND_PASS_COUNT:-0}
-              --frontend-warn-count ${ZAP_FRONTEND_WARN_COUNT:-0}
-              --frontend-fail-count ${ZAP_FRONTEND_FAIL_COUNT:-0}
-              --project-slug $PROJECT_SLUG
-            )
             # Evaluate the full command before passing it in so it doesn't
             # get improperly interpolated by Cloud.gov.
-            CMD="python manage.py process_owasp_scan ${CMD_ARGS[@]}"
+            CMD="python manage.py process_owasp_scan $CIRCLE_BUILD_NUM --backend-pass-count ${ZAP_BACKEND_PASS_COUNT:-0} --backend-warn-count ${ZAP_BACKEND_WARN_COUNT:-0} --backend-fail-count ${ZAP_BACKEND_FAIL_COUNT:-0} --frontend-pass-count ${ZAP_FRONTEND_PASS_COUNT:-0} --frontend-warn-count ${ZAP_FRONTEND_WARN_COUNT:-0} --frontend-fail-count ${ZAP_FRONTEND_FAIL_COUNT:-0} --project-slug $PROJECT_SLUG"
```
Comment (andrew-jameson) [inline .circleci/owasp/jobs.yml]: If you got it to work, I'm not sure I want to mess with it but can we improve formatting/readability? If some newlines break it, it's probably not worth the effort to get it "perfect".

**Review by andrew-jameson — APPROVED** [review]
Had some minor questions but regardless of outcome, I'm good with the changes in `jobs.yml`. Will need to revert `workflows.yml` hijacking per the TODO. Great work, @elipe17 , thank you!

**Review by jtimpe — APPROVED** [review]

File: .circleci/build-and-test/workflows.yml [hunk 3]
```diff
@@ -3,15 +3,9 @@
...
-      - test-backend:
-          requires:
-            - secrets-check      
-      - test-frontend:
-          requires:
-            - secrets-check
-      - test-e2e:
-          requires:
-            - secrets-check
+      # TODO: Revert back to tests after review
```
Comment (ADPennington) [inline .circleci/build-and-test/workflows.yml]: @elipe17 this PR is ready to revert changes needed for testing. build-and-test-all workflow was successful. Please ping me for qasp approval after the revert. 

zaphotfix

Comment (elipe17) [reply .circleci/build-and-test/workflows.yml]: @ADPennington I have reverted the workflow to it's original state

**Review by ADPennington — APPROVED** [review]
🚀

**Pull request merged**